### PR TITLE
Added function to get items from the shared space

### PR DESCRIPTION
**Pull request by lodesmets**
Home assistant issue: https://github.com/home-assistant/core/issues/92600
This person used the shared space. I did some testing and found the API endpoint to get the items in the shared space

### CODE REVIEW

**Comment by lodesmets** [comment]
> Hi @lodesmets thanks for this addition, but i've left some small comment

I made the change to the branch. lodesmets:shared_photos.

But for some reason it doesn't show up in the PR for me. 
I'm guessing a github bug. If it didn't change in a couple days, I will close this, and open a new PR

**Comment by mib1185** [comment]
not sure what was the problem, but just close and re-open solved it 😆

File: src/synology_dsm/api/photos/__init__.py [hunk 1]
```diff
@@ -75,6 +76,44 @@ async def get_items_from_album(
...
+                    item["id"],
+                    item["type"],
+                    item["filename"],
+                    item["filesize"],
+                    item["additional"]["thumbnail"]["cache_key"],
+                    size,
+                    item["owner_user_id"] == 0,
+                )
+            )
+        return items
```
Comment (mib1185) [inline src/synology_dsm/api/photos/__init__.py:115]: i think we should move this part in a separate function and reuse it in `get_items_from_album`, `get_items_from_shared_space` and `get_items_from_search`

**Review by mib1185 — CHANGES_REQUESTED** [review]
Hi @lodesmets 
just one other point left, we have 3x the same code 🙈

**Review by mib1185 — APPROVED** [review]
LGTM, thx @lodesmets 👍

**Pull request merged**